### ISSUE TITLE
Document Poetry dependency schema

### ISSUE THREAD

**Issue by madkinsz** [python-poetry/poetry#6600]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.

## Issue

Hi! I can't seem to find documentation on the all the available options for a Poetry dependency. It'd be great to have a schema somewhere in the documentation.

For example, `platform` (added in https://github.com/python-poetry/poetry/commit/20c87bc0e32d26f406576b9819b4078e80ab79d9) is not included in the dependency specification page and is not PEP-0508 compliant so users and consumers (e.g. https://github.com/pdm-project/pdm/issues/1384) do not know what fields can be used.

https://python-poetry.org/docs/dependency-specification seems like the relevant page to add this information

Related to https://github.com/python-poetry/poetry/issues/738, https://github.com/python-poetry/poetry/pull/1207

**Comment by dimbleby** [comment]
https://github.com/python-poetry/poetry-core/blob/main/src/poetry/core/json/schemas/poetry-schema.json

If you can find a sensible place to point at that from the docs I expect an MR would be welcome

**Comment by madkinsz** [comment]
Thanks for the link! That's helpful. I'm a bit hesitant to just link out to `main` and I don't see a great way to template a tag when the documentation is generated. Seems like that'd be important for maintenance over time.

**Comment by dimbleby** [comment]
I agree.  Maybe put something together anyway and let the maintainers advise what to do about it.

The schema would so clearly be targeted at developers only: perhaps it's not crazy to tuck this away in a footnote and say something like "there's a schema in the poetry-core repository if you really want to get into this, you'll have to figure out for yourself what the right tag is".
